src/org/usfirst/frc2130/visionTargetting/Robot.java

```diff
@@ -0,0 +1,263 @@
+ * documentation. If you change the name of this class or the package after
+ * creating this project, you must also update the manifest file in the resource
+ * directory.
+ */
+
+
+
+public class Robot extends IterativeRobot {
+	
+
+
+	NetworkTable table;
+
+    Command autonomousCommand;
+    
+	public VisionThread visionThread;
+	
+	public static  Object imgLock;
+	
+	public static double centerX;
+	
+	public static double run;
+	
+
+	private static final int IMG_WIDTH = 320;
+	private static final int IMG_HEIGHT = 240;
+
+	private boolean run1;
+    
+    public static OI oi;
+    // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+    public static robotDrive robotDrive;
+
+    // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=DECLARATIONS
+
+    /**
+     * This function is run when the robot is first started up and should be
+     * used for any initialization code.
+     */
+    @Override
+    public void robotInit() {
+    RobotMap.init();  
+    	
+    // Creates UsbCamera and MjpegServer [1] and connects them
+
+    CameraServer.getInstance().startAutomaticCapture();
+
+    // Creates the CvSink and connects it to the UsbCamera
+    CvSink cvSink = CameraServer.getInstance().getVideo();
+
+    // Creates the CvSource and MjpegServer [2] and connects them
+    CvSource outputStream = CameraServer.getInstance().putVideo("Blur", 640, 480); 
+       
+         
+        // BEGIN AUTOGENERATED CODE, SOURCE = ROBOTBUILDER ID = CONSTRUCTORS
+        robotDrive = new robotDrive();
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID = CONSTRUCTORS
+        // OI must be constructed after subsystems. If the OI creates Commands
+        //(which it very likely will), subsystems are not guaranteed to be
+        // constructed yet. Thus, their requires() statements may grab null
+        // pointers. Bad news. Don't move it.
+        oi = new OI();
+
+        // instantiate the command used for the autonomous period
+        // BEGIN AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
+
+        autonomousCommand = new AutonomousCommand();
+
+        // END AUTOGENERATED CODE, SOURCE=ROBOTBUILDER ID=AUTONOMOUS
+        
+        CameraServer.getInstance().startAutomaticCapture();
+        UsbCamera camera = new UsbCamera("USB Camera", 0);
+        camera.setResolution(IMG_WIDTH, IMG_HEIGHT);
+        
+        CameraServer.getInstance().getVideo();
+       
+        new Thread(() -> {        
+            CvSink cvSink = CameraServer.getInstance().getVideo();
+            CvSource outputStream = CameraServer.getInstance().putVideo("Blur", 640, 480);
+            
+            Mat source = new Mat();
+            Mat output = new Mat();
+            
+            while(!Thread.interrupted()) {
+                cvSink.grabFrame(source);
+                Imgproc.cvtColor(source, output, Imgproc.COLOR_BGR2GRAY);
+                outputStream.putFrame(output);
+            }
```

Comment:
Lines 162-172 are implementing one of the ScreenStepsLive examples. That's great, but probably not what's intended for this program.  Most importantly:

`while(!Thread.interrupted()) {
...
`

means that the block of code in the middle will run until the thread is interrupted by the computer (probably when disabling). After it is interrupted it will proceed to the blocks of code below.  I think this should be removed in favor of the pipeline.

There is an interesting task that is being performed here, which is to take the original color image and convert it to black and white, then transmit it to the named stream.  If you want to view the real time output of your filter pipeline, something similar will need to be done.  I'll add an example in the updated code.